PWGHF/Core/HfMlResponseLcToPKPi.h

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HfMlResponseLcToPKPi.h
+/// \brief Class to compute the ML response for Lc+ → p K- π+ analysis selections
+/// \author Grazia Luparello <grazia.luparello@cern.ch>
+
+#ifndef PWGHF_CORE_HFMLRESPONSELCTOPKPI_H_
+#define PWGHF_CORE_HFMLRESPONSELCTOPKPI_H_
+
+#include <vector>
+
+#include "PWGHF/Core/HfMlResponse.h"
+
+// Fill the map of available input features
+// the key is the feature's name (std::string)
+// the value is the corresponding value in EnumInputFeatures
+#define FILL_MAP_LC(FEATURE)                                         \
+  {                                                                  \
+#FEATURE, static_cast < uint8_t>(InputFeaturesLcToPKPi::FEATURE) \
+  }
+
+// Check if the index of mCachedIndices (index associated to a FEATURE)
+// matches the entry in EnumInputFeatures associated to this FEATURE
+// if so, the inputFeatures vector is filled with the FEATURE's value
+// by calling the corresponding GETTER from OBJECT
+#define CHECK_AND_FILL_VEC_LC_FULL(OBJECT, FEATURE, GETTER)    \
+  case static_cast<uint8_t>(InputFeaturesLcToPKPi::FEATURE): { \
+    inputFeatures.emplace_back(OBJECT.GETTER());               \
+    break;                                                     \
+  }
+
+// Specific case of CHECK_AND_FILL_VEC_LC_FULL(OBJECT, FEATURE, GETTER)
+// where OBJECT is named candidate and FEATURE = GETTER
+#define CHECK_AND_FILL_VEC_LC(GETTER)                         \
+  case static_cast<uint8_t>(InputFeaturesLcToPKPi::GETTER): { \
+    inputFeatures.emplace_back(candidate.GETTER());           \
+    break;                                                    \
+  }
+
+// Variation of CHECK_AND_FILL_VEC_LC_FULL(OBJECT, FEATURE, GETTER)
+// where GETTER is a method of hfHelper
+#define CHECK_AND_FILL_VEC_LC_HFHELPER(OBJECT, FEATURE, GETTER) \
+  case static_cast<uint8_t>(InputFeaturesLcToPKPi::FEATURE): {  \
+    inputFeatures.emplace_back(hfHelper.GETTER(OBJECT));        \
+    break;                                                      \
+  }
```

Comment:
The names of these macros in principle could clash with the ones defined for the Lc->pK0s channel https://github.com/AliceO2Group/O2Physics/blob/master/PWGHF/Core/HfMlResponseLcToK0sP.h#L30-L59